Lib/fontTools/otlLib/builder.py

```diff
@@ -589,7 +589,10 @@ def __init__(self, useClass0):
         self.useClass0_ = useClass0
 
     def canAdd(self, glyphs):
-        glyphs = tuple(glyphs)
+        if isinstance(glyphs, (set, frozenset)):
+            glyphs = tuple(sorted(glyphs))
+        elif not isinstance(glyphs, tuple):
```

Comment:
Since tuples are immutable, if x is already a tuple, then "tuple(x)" returns x itself.  So you don't need to specialcase it.  You can simply write:
```
+        if isinstance(glyphs, (set, frozenset)):
+            glyphs = tuple(sorted(glyphs))
+        glyphs = tuple(glyphs)